### PR TITLE
[SPARK-32507][DOCS][PYTHON] Add main page for PySpark documentation

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to write the main page of PySpark documentation. The base work is finished at https://github.com/apache/spark/pull/29188.

### Why are the changes needed?

For better usability and readability in PySpark documentation.

### Does this PR introduce _any_ user-facing change?

Yes, it creates a new main page as below:

Screen Shot 2020-07-31 at 10 02 44 PM

### How was this patch tested?

Manually built the PySpark documentation.

```bash
cd python
make clean html
```

### CODE REVIEW

**Review by viirya — COMMENTED** [review]
main package? main page? I mean the PR title.

File: python/docs/source/index.rst [hunk 1]
```diff
@@ -21,8 +21,42 @@
...
+**MLlib**
+
+Built on top of Spark, MLlib is a scalable machine learning library that provides
+a uniform set of high-level APIs that help users create and tune practical machine
+learning pipelines.
+
+**Spark Core**
+
+Spark Core is the underlying general execution engine for the Spark platform that all
+other functionality is built on top of. It provides an RDD (Resilient Disributed Data)
```
Comment (srowen) [inline python/docs/source/index.rst:54]: Data -> Dataset

File: python/docs/source/index.rst [hunk 2]
```diff
@@ -21,8 +21,42 @@
 PySpark Documentation
 =====================
 
+PySpark is an interface for Apache Spark in Python language. It not only offers for you
```
Comment (srowen) [inline python/docs/source/index.rst:24]: in the Python language, or just "in Python"

File: python/docs/source/index.rst [hunk 3]
```diff
@@ -21,8 +21,42 @@
...
+to write an application in the Python APIs but also provides PySpark shell so you can
+interactively analyze your data in a distributed environment. PySpark supports most
+of Spark features such as Spark SQL, DataFrmae, Streaming, MLlib
```
Comment (srowen) [inline python/docs/source/index.rst:27]: most of Spark's features

File: python/docs/source/index.rst [hunk 4]
```diff
@@ -21,8 +21,42 @@
 PySpark Documentation
 =====================
 
+PySpark is an interface for Apache Spark in Python language. It not only offers for you
+to write an application in the Python APIs but also provides PySpark shell so you can
```
Comment (srowen) [inline python/docs/source/index.rst:25]: Maybe "It not only allows you to write Spark applications using Python APIs, but also provides the PySpark shell for interactively analyzing ..."

**Review by srowen — COMMENTED** [review]
Looks nice

**Review by nchammas — COMMENTED** [review]
What's `docs/img/pyspark-components.pptx` for?

**Comment by HyukjinKwon** [comment]
> What's docs/img/pyspark-components.pptx for?

It is for the image I used in the main page in case some people want to edit. There are other pptx files in `docs/img` as well for that purpose.

File: python/docs/source/index.rst [hunk 5]
```diff
@@ -21,8 +21,42 @@
 PySpark Documentation
 =====================
 
+PySpark is an interface for Apache Spark in Python. It not only allows you to write
+Spark applications using Python APIs, but also provides the PySpark shell for
+interactively analyzing your data in a distributed environment. PySpark supports most
+of Spark's features such as Spark SQL, DataFrmae, Streaming, MLlib
```
Comment (viirya) [inline python/docs/source/index.rst:27]: DataFrmae -> DataFrame

File: python/docs/source/index.rst [hunk 6]
```diff
@@ -21,8 +21,42 @@
...
+**MLlib**
+
+Built on top of Spark, MLlib is a scalable machine learning library that provides
+a uniform set of high-level APIs that help users create and tune practical machine
+learning pipelines.
+
+**Spark Core**
+
+Spark Core is the underlying general execution engine for the Spark platform that all
+other functionality is built on top of. It provides an RDD (Resilient Disributed Dataset)
```
Comment (viirya) [inline python/docs/source/index.rst:54]: Disributed -> Distributed

**Review by viirya — COMMENTED** [review]
This looks good. I just found a few typos.

**Review by viirya — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Thank you @viirya for approaching this. I am merging this to master.

**Comment by BryanCutler** [comment]
Looks great, I think it will be very helpful for PySpark to have it's own main page. Thanks @HyukjinKwon !

**Pull request closed without merging**